src/libraries/System.Private.CoreLib/src/System/Text/Ascii.cs

```diff
@@ -61,58 +59,57 @@ public static partial class Ascii
                         // Process byte inputs with lengths [4, 7]
                         return AllBytesInUInt32AreAscii(
                             Unsafe.ReadUnaligned<uint>(ref Unsafe.As<T, byte>(ref searchSpace)) |
-                            Unsafe.ReadUnaligned<uint>(ref Unsafe.As<T, byte>(ref Unsafe.Subtract(ref searchSpaceEnd, sizeof(uint)))));
+                            Unsafe.ReadUnaligned<uint>(ref Unsafe.As<T, byte>(ref Unsafe.Add(ref searchSpace, length - sizeof(uint)))));
                     }
 
                     // Process inputs with lengths [0, 3]
-                    while (Unsafe.IsAddressLessThan(ref searchSpace, ref searchSpaceEnd))
+                    for (int j = 0; j < length; j++)
                     {
                         if (typeof(T) == typeof(byte)
-                            ? (Unsafe.BitCast<T, byte>(searchSpace) > 127)
-                            : (Unsafe.BitCast<T, char>(searchSpace) > 127))
+                            ? (Unsafe.BitCast<T, byte>(Unsafe.Add(ref searchSpace, j)) > 127)
```

Comment:
`j` should be a native integer in order to avoid the sign-extending moves.
See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4aNYgGYmpBkIYBvGgzNNdwAJ4YYDALIAKWADMG12wzEYoaBrx5/ABMEAEoGAF4APgYAVS5cbBdBAEEgoOcYN29fYLCNanMLdxs7J1cSzxy/Lg4AjDzw6LiEpNT0zOyfP14Q0IKAXyA===)